central/policy/service/service_impl.go

```diff
@@ -314,12 +314,26 @@ func (s *serviceImpl) PatchPolicy(ctx context.Context, request *v1.PatchPolicyRe
 	return s.PutPolicy(ctx, policy)
 }
 
-// DeletePolicy deletes an policy from the system.
+// DeletePolicy deletes a policy from the system.
 func (s *serviceImpl) DeletePolicy(ctx context.Context, request *v1.ResourceByID) (*v1.Empty, error) {
 	if request.GetId() == "" {
 		return nil, errors.Wrap(errox.InvalidArgs, "A policy id must be specified to delete a Policy")
 	}
 
+	policy, exists, err := s.policies.GetPolicy(ctx, request.GetId())
+	if !exists {
+		// make repeated calls with the same policy ID idempotent
+		return &v1.Empty{}, nil
```

Comment:
I took a look at the datastore implementation to see if this matches the existing behavior, and it looks like it does, assuming sql deletes that delete zero rows does not return a sql error (afair it does not).

Reply:
After I wrote that comment I realized it would be easier to just test out an idempotent DELETE operation against staging.  It is, in fact, idempotent!